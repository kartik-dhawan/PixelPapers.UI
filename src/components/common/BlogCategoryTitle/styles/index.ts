import { GLOBAL_COLORS, PAGE_LEVEL_COLORS } from "@/utils/colorSchemes"

export const styles = {
  // blog category heading section
  blogCategoryTitle: {
    margin: "1rem 0rem 0rem",
    fontSize: "72px",
    textTransform: "uppercase",
    color: GLOBAL_COLORS.TEXT_PRIMARY_DARK,
  },
  blogCategoryTitleSubtext: {
    maxWidth: "720px",
    textAlign: "center",
    fontSize: "18px",
    color: PAGE_LEVEL_COLORS.TRAVEL.TEXT_PRIMARY_LIGHTER,
  },

  // curatedBy component
  curatedByColorBox: {
    height: "32px",
    aspectRatio: "1/1",
    backgroundColor: PAGE_LEVEL_COLORS.TRAVEL.BACKGROUND_PRIMARY_DARK,
    borderRadius: "50%",
    margin: "0rem 6px 0rem 12px",
  },
  curatedByUserLink: {
    "& a": {
      fontSize: "22px",
      color: "#111",
      fontWeight: 700,
      textDecoration: "none",
    },
  },
  // curatedBy
}
