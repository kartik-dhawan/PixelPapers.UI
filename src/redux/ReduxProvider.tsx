"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
