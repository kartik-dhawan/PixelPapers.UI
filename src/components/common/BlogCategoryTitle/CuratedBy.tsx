"use client"

import { Box } from "@mui/material"
import Link from "next/link"
import { styles } from "./styles"

interface CuratedByProps {
  name: string
  url?: string
}

const CuratedBy = ({ name, url }: CuratedByProps) => {
  return (
    <Box sx={{ display: "flex", gap: "4px", alignItems: "center" }}>
      <Box component="span" sx={{ fontSize: "22px", fontWeight: 400 }}>
        Curated
      </Box>
      <Box component="span" sx={{ fontSize: "14px" }}>
        by
      </Box>
      <Box component="span" sx={styles.curatedByColorBox} />
      <Box component="span" sx={styles.curatedByUserLink}>
        {url ? (
          <Link href={url} target="_blank">
            {name}
          </Link>
        ) : (
          <>{name}</>
        )}
      </Box>
    </Box>
  )
}

export default CuratedBy
