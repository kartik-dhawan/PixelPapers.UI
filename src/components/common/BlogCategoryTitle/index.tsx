"use client"

import { RootType } from "@/redux/store"
import { dmSans } from "@/utils/fonts"
import { Box, Stack } from "@mui/material"
import { useSelector } from "react-redux"
import CuratedBy from "./CuratedBy"
import { styles } from "./styles"
import { CURATED_BY_LINK_DEFAULT_TEXT } from "@/utils/constants"

interface BlogCategoryTitleProps {
  category: "travel"
  themeColor: string
}

const BlogCategoryTitle = ({
  category,
  themeColor,
}: BlogCategoryTitleProps) => {
  const { content } = useSelector((state: RootType) => state.contentSlice)

  const categoryData = {
    travel: {
      title: content?.travelPageTitle,
      description: content?.travelPageTitleSubtext,
    },
  }

  return (
    <Stack
      className={`${category}PageHeadWrapper ${dmSans.className}`}
      data-testid={`${category}PageHeadWrapper`}
      alignItems="center"
      margin={{ xs: "2rem 0rem", md: "4rem 0rem" }}
    >
      <CuratedBy
        name={CURATED_BY_LINK_DEFAULT_TEXT}
        url={content?.curatedByLink}
        themeColor={themeColor}
      />
      {content && (
        <>
          <Box component="h1" sx={styles.blogCategoryTitle}>
            {categoryData[category]?.title}
          </Box>
          <Box component="p" sx={styles.blogCategoryTitleSubtext}>
            {categoryData[category]?.description}
          </Box>
        </>
      )}
    </Stack>
  )
}

export default BlogCategoryTitle
