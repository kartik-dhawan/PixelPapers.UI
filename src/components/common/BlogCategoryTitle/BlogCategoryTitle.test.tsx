import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import BlogCategoryTitle from "."

describe("Blog category title section component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <BlogCategoryTitle category="travel" />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
