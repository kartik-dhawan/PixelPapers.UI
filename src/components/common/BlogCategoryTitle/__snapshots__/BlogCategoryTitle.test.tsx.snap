// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog category title section component -- should match the snapshot 1`] = `
<div
  className="MuiStack-root travelPageHeadWrapper className css-18a369j-MuiStack-root"
  data-testid="travelPageHeadWrapper"
>
  <div
    className="MuiBox-root css-wpwytb"
  >
    <span
      className="MuiBox-root css-vuei1s"
    >
      Curated
    </span>
    <span
      className="MuiBox-root css-1kuy7z7"
    >
      by
    </span>
    <span
      className="MuiBox-root css-5us4vj"
    />
    <span
      className="MuiBox-root css-eyghm8"
    >
      kartik-dhawan
    </span>
  </div>
  <h1
    className="MuiBox-root css-heu39z"
  />
  <p
    className="MuiBox-root css-zduu3p"
  />
</div>
`;
